test/visual/visual.js

```diff
@@ -138,24 +149,30 @@ module.exports = {
                         `${currentDir}/${type}/${view}.png`
                     );
                     const fileSizeInBytes = stats.size;
-                    console.log(
-                        `📸 ${view}.png => ${fileSizeInBytes} bytes, ${percentDiff}% different`
-                    );
 
-                    if (numDiffPixels > 10) {
-                        diff.pack().pipe(
-                            fs.createWriteStream(
-                                `${currentDir}/${view}-diff.png`
-                            )
-                        );
-                    }
-                    // A number of comparisons come out to 10 inexplicable pixes EVERY time,
-                    // so let them have their stupid 10 pixels!
-                    expect(
-                        numDiffPixels,
-                        'number of different pixels'
-                    ).be.lessThan(11);
-                    resolve();
+                    resolve({
+                        // Return the diff file creation and file comparisin expect so that the "test"
+                        // can be run only when it "passes" or is the second try.
+                        test: () => {
+                            console.log(
+                                `📸 ${view}.png => ${fileSizeInBytes} bytes, ${percentDiff}% different`
+                            );
+                            if (numDiffPixels > 10) {
+                                diff.pack().pipe(
+                                    fs.createWriteStream(
+                                        `${currentDir}/${view}-diff.png`
+                                    )
+                                );
+                            }
+                            // A number of comparisons come out to 10 inexplicable pixes EVERY time,
+                            // so let them have their stupid 10 pixels!
+                            expect(
+                                numDiffPixels,
+                                'number of different pixels'
+                            ).be.lessThan(11);
```

Comment:
wasn't aware that we had this threshold here before, do we have any idea what this 10 pixel discrepancy is coming from?

Reply:
Welp, I don't know when it stopped, but I can't seem to trigger it anymore. Maybe something with an upgrade to puppeteer since I originally set this up? It's looking like Ben's note above _might_ also play out, and if so I'll bring both of these notes together in a new PR 🤞